File: codewars/06-convert_string_to_camelcase/convert.py
def to_camel_case(text):
    new_text = ""
    for i in range(len(text)):
        if text[i-1] == "-" or text[i-1] == "_":
            continue
        if text[i] != "-" and text[i] != "_":
            new_text += text[i]
        else:
            new_text += text[i+1].upper()
    return new_text


def to_camel_case(text):
    return text[:1] + text.title()[1:].replace('_', '').replace('-', '')


def to_camel_case(text):
    removed = text.replace('-', ' ').replace('_', ' ').split()
    if len(removed) == 0:
        return ''
    return removed[0]+ ''.join([x.capitalize() for x in removed[1:]])


import re
def to_camel_case(text):
    return re.sub('[_-](.)', lambda x: x.group(1).upper(), text)


def to_camel_case(text):
    words = text.replace('_', '-').split('-')
    return words[0] + ''.join([x.title() for x in words[1:]])

if __name__ ==  "__main__":
    print(to_camel_case("hola-mundo_maicra"))

File: codewars/06-convert_string_to_camelcase/convert.cpp
#include <iostream>
#include <string>

std::string to_camel_case(std::string text) {
  std::string upperCamelCase = "";
  for (int i = 0; i < text.size(); i++) {
    if (text[i - 1] == '_' | text[i - 1] == '-') continue;
    if (text[i] != '_' & text[i] != '-') {
      upperCamelCase += text[i];
    } else {
      upperCamelCase += toupper(text[i + 1]);
    }
  }
  return upperCamelCase;
}

std::string to_camel_case(std::string text) {
  for (int i = 0; i < text.size(); ++i) {
    if (text[i] == '-' || text[i] == '_') {
      text.erase(i, 1);
      text[i] = toupper(text[i]);
    }
  }
  return text;
}

std::string to_camel_case(std::string s) {
  for (int i{0}; i < s.size(); ++i)
    if (s[i] == '-' || s[i] == '_') s.erase(i, 1), s[i] = toupper(s[i]);
  return s;
}

#include <cctype>

std::string to_camel_case(std::string text) {
  for (int i = 0; i < text.size(); i++)
    if (text[i] == '-' || text[i] == '_')
      text.replace(i, 2, 1, toupper(text[i + 1]));
  return text;
}

int main() {
  std::cout << to_camel_case("hola-mundo_maicra") << "\n";

  return 0;
}